scripts/install

```diff
@@ -1,96 +1,140 @@
-sudo mysql -u root -p$(sudo cat /etc/yunohost/mysql) < ../conf/create_db.sql
-sudo sed -i "s@phpmyadmin@$db_user@g" $final_path/sql/create_tables.sql
-sudo mysql -u $db_user -p$db_pwd < $final_path/sql/create_tables.sql
-
-# Configuration
-echo "Configuring application..."
-sed -i "s@YNH_DOMAIN@$domain@g" ../conf/config.inc.php
-sed -i "s@YNH_PMA_USER@$db_user@g" ../conf/config.inc.php
-sed -i "s@YNH_PMA_PASSWORD@$db_pwd@g" ../conf/config.inc.php
-sed -i "s@YNH_MYSQL_ROOT_PASSWORD@$(sudo cat /etc/yunohost/mysql)@g" ../conf/config.inc.php
-sudo cp ../conf/config.inc.php $final_path
-
-# Files owned by root, www-data can just read
-echo "Setting permission..."
-sudo chown -R root: $final_path
-sudo find $final_path -type f -exec chmod 644 {} \;
-sudo find $final_path -type d -exec chmod 755 {} \;
+#=================================================
+# STANDARD MODIFICATIONS
+#=================================================
+# CREATE A MYSQL DATABASE
+#=================================================
+
+db_name=$(ynh_sanitize_dbid $app)
+ynh_app_setting_set $app db_name $db_name
+ynh_mysql_setup_db $db_name $db_name
+
+#=================================================
+# DOWNLOAD, CHECK AND UNPACK SOURCE
+#=================================================
+
+ynh_app_setting_set $app final_path $final_path
+# Download, check integrity, uncompress and patch the source from app.src
+ynh_setup_source "$final_path"
+
+#=================================================
+# NGINX CONFIGURATION
+#=================================================
+
+# Create a dedicated nginx config
+ynh_add_nginx_config
+
+#=================================================
+# CREATE DEDICATED USER
+#=================================================
+
+# Create a system user
+ynh_system_user_create $app
+
+#=================================================
+# PHP-FPM CONFIGURATION
+#=================================================
+
+# Create a dedicated php-fpm config
+ynh_add_fpm_config
+
+#=================================================
+# SPECIFIC SETUP
+#=================================================
+# POPULATE THE DATABASE
+#=================================================
+
+ynh_replace_string "YNH_PMA_USER" "$db_name" ../conf/create_db.sql
+ynh_mysql_connect_as "$db_name" "$db_pwd" "$db_name" \
+    < ../conf/create_db.sql
+ynh_replace_string "phpmyadmin" "$db_name" $final_path/sql/create_tables.sql
+ynh_mysql_connect_as "$db_name" "$db_pwd" "$db_name" \
+    < $final_path/sql/create_tables.sql
+
+#=================================================
+# CONFIGURE PHPMYADMIN
+#=================================================
+
+ynh_replace_string "YNH_DOMAIN" "$domain" ../conf/config.inc.php
+ynh_replace_string "YNH_PMA_USER" "$db_name" ../conf/config.inc.php
+ynh_replace_string "YNH_PMA_PASSWORD" "$db_pwd" ../conf/config.inc.php
+ynh_replace_string "YNH_MYSQL_ROOT_PASSWORD" "$(cat $MYSQL_ROOT_PWD_FILE)" ../conf/config.inc.php
```

Comment:
`MYSQL_ROOT_PWD_FILE` env var hasn't been defined.

Reply:
`MYSQL_ROOT_PWD_FILE` should be defined by the helper script sql in helper.d
Is it not set?

Reply:
Indeed! I didn't know that :astonished: